### ISSUE TITLE
Port supplied with 

### ISSUE THREAD

**Issue by kielstr** [rustyconover/net-amazon-s3#88]
Any port supplied via the vender is dropped in the following code

```
 # Ensure Host header is present as its required
    if (!$req->header('host')) {
        $req->header('Host' => $req->uri->host);
    }
```

```
    vendor =>  Net::Amazon::S3::Vendor::Generic->new (
            host                 => 's3.example.com:9999',
            use_https            => 0,
            use_virtual_host     => 0,
            authorization_method => 'Net::Amazon::S3::Signature::V4',
            default_region       => '<region>',
        ),
```

https://github.com/rustyconover/net-amazon-s3/blob/e72c1b1aeb5f26f669ba5a2662bc694ae205f542/lib/Net/Amazon/S3/Signature/V4Implementation.pm#L173-L176

**Comment by kielstr** [comment]
This invalidates the signed URI when the server recalculates it.

**Comment by happy-barney** [comment]
Thanks for reporting, fixed in v0.96 (I hope, still developing without real service on hand)
